### PR TITLE
feat: 1689 charge message persistence functions

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

Step two in the _prerequisite tasks_ of #1689:
- [ ] Two new functions in the Charges information and prices command flows: Reacts to ChargeInformation/Prices accepted events and persists message (document) id, with reference to the charge

## References

* #1689

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/ChargeInformationMessagePersisterEndpointConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using GreenEnergyHub.Charges.Infrastructure.Core.MessagingExtensions.Serialization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GreenEnergyHub.Charges.FunctionHost.Configuration
+{
+    internal static class ChargeInformationMessagePersisterEndpointConfiguration
+    {
+        internal static void ConfigureServices(IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddScoped<JsonMessageDeserializer<ChargePriceOperationsAcceptedEvent>>();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/ChargeInformationMessagePersisterEndpointConfiguration.cs:26]:
```suggestion
            serviceCollection.AddScoped<JsonMessageDeserializer<ChargeInformationOperationsAcceptedEvent>>();
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/ChargeInformationMessagePersisterEndpointTests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+            [Theory]
+            [InlineAutoMoqData]
+            public async Task RunAsync_WhenChargeInformationOperationsAcceptedEvent_ChargeMessageIsPersisted(
+                ChargeInformationOperationsAcceptedEventBuilder chargeInformationOperationsAcceptedEventBuilder)
+            {
+                // Arrange
+                await using var chargesDatabaseReadContext = _databaseManager.CreateDbContext();
+
+                var operationsConfirmedEvent = chargeInformationOperationsAcceptedEventBuilder.Build();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/ChargeInformationMessagePersisterEndpointTests.cs:68]:
```suggestion
                var operationsAcceptedEvent = chargeInformationOperationsAcceptedEventBuilder.Build();
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/ChargePriceMessagePersisterEndpointTests.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+            [Theory]
+            [InlineAutoMoqData]
+            public async Task RunAsync_WhenChargePriceOperationsAcceptedEvent_ChargeMessageIsPersisted(
+                ChargePriceOperationsAcceptedEventBuilder chargePriceOperationsAcceptedEventBuilder)
+            {
+                // Arrange
+                await using var chargesDatabaseReadContext = _databaseManager.CreateDbContext();
+
+                var operationsConfirmedEvent = chargePriceOperationsAcceptedEventBuilder.Build();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/ChargePriceMessagePersisterEndpointTests.cs:68]:
```suggestion
                var operationsAcceptedEvent = chargePriceOperationsAcceptedEventBuilder.Build();
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs [hunk 4]
```diff
@@ -50,7 +49,15 @@ public ChargeMessageRepositoryTests(ChargesDatabaseFixture fixture)
             // Arrange
             await using var chargesDatabaseWriteContext = _databaseManager.CreateDbContext();
             var charge = await GetCharge(chargesDatabaseWriteContext);
-            var chargeMessage = chargeMessageBuilder.WithChargeId(charge.Id).Build();
+            var chargeIdentifier = await GetChargeIdentifier(
+                chargesDatabaseWriteContext,
+                charge,
+                SeededData.MarketParticipants.SystemOperator.Gln);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs:55]: Is this needed?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Application.Charges.Handlers.ChargeInformation
+{
+    [UnitTest]
+    public class ChargeInformationMessagePersisterTests
+    {
+        [Theory]
+        [InlineAutoDomainData]
+        public async Task HandleAsync_WhenCalled_ShouldCallChargeMessageRepository(
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs:34]:
```suggestion
        public async Task PersistMessageAsync_WhenCalled_ShouldCallChargeMessageRepository(
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+using GreenEnergyHub.Charges.TestCore.Builders.Command;
+using GreenEnergyHub.TestHelpers;
+using Moq;
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Application.Charges.Handlers.ChargeInformation
+{
+    [UnitTest]
+    public class ChargeInformationMessagePersisterTests
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs:30]: Wanna cover the null exception with a test?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Application.Charges.Handlers.ChargePrice
+{
+    [UnitTest]
+    public class ChargePriceMessagePersisterTests
+    {
+        [Theory]
+        [InlineAutoDomainData]
+        public async Task HandleAsync_WhenCalled_ShouldCallChargeMessageRepository(
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs:34]:
```suggestion
        public async Task PersistMessageAsync_WhenCalled_ShouldCallChargeMessageRepository(
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+using GreenEnergyHub.Charges.TestCore.Builders.Command;
+using GreenEnergyHub.TestHelpers;
+using Moq;
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Application.Charges.Handlers.ChargePrice
+{
+    [UnitTest]
+    public class ChargePriceMessagePersisterTests
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs:30]: Wanna cover the null exception with a test?

**Review by prtandrup — APPROVED** [review]
Approved with a few comments

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs (hunk 4 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs:55]: Not absolutely necessary, its just to make sure we have the marketparticipant in the db. Should we remove it?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs (hunk 6 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargeInformation/ChargeInformationMessagePersisterTests.cs:32]: Absolutely!

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs (hunk 8 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePrice/ChargePriceMessagePersisterTests.cs:32]: Indeed!

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs (hunk 4 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeMessageRepositoryTests.cs]: Removed

**Pull request merged**